### PR TITLE
Integrate the Hikari metrics to Prometheus by java agent

### PR DESCRIPTION
**Pull request by dachuan9e**
#11304 

Integrate the Hikari metrics to Prometheus by java agent

### CODE REVIEW

**Comment by dachuan9e** [comment]
Name the hikari pool as the following format for identify when exporting metrics data.

  <schema_name>/<datasource_name>
  
 example:  sharding_db/ds_0

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/config/datasource/JDBCParameterDecoratorHelper.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+package org.apache.shardingsphere.infra.config.datasource;
+
+import org.apache.shardingsphere.infra.spi.ShardingSphereServiceLoader;
+
+import javax.sql.DataSource;
+import java.util.Optional;
+
+public class JDBCParameterDecoratorHelper {
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/config/datasource/JDBCParameterDecoratorHelper.java:25]: If it is just a util class, it is better to define it as a final class and add `@NoArgsConstructor(access = AccessLevel.PRIVATE)` for it.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/datasource/factory/JDBCRawBackendDataSourceFactory.java [hunk 2]
```diff
@@ -22,28 +22,28 @@
...
  * Backend data source factory using {@code HikariDataSource} for JDBC raw.
  */
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 @Slf4j
 public final class JDBCRawBackendDataSourceFactory implements JDBCBackendDataSourceFactory {
-    
+
     private static final JDBCRawBackendDataSourceFactory INSTANCE = new JDBCRawBackendDataSourceFactory();
-    
+
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/datasource/factory/JDBCRawBackendDataSourceFactory.java:42]: Please reverse these invalid changes.

Comment (tristaZero) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/datasource/factory/JDBCRawBackendDataSourceFactory.java:42]: [Settings for IDEA](https://shardingsphere.apache.org/community/data/shardingsphere-settings.jar) FYI.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/datasource/factory/JDBCRawBackendDataSourceFactory.java (hunk 2 above)
Comment (dachuan9e) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/datasource/factory/JDBCRawBackendDataSourceFactory.java:42]: ok

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/config/datasource/JDBCParameterDecoratorHelper.java (hunk 1 above)
Comment (dachuan9e) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/config/datasource/JDBCParameterDecoratorHelper.java:25]: changed

File: shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-metrics/shardingsphere-agent-metrics-api/src/test/java/org/apache/shardingsphere/agent/metrics/api/advice/DataSourceAdviceTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.lang.reflect.Method;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DataSourceAdviceTest {
```
Comment (tristaZero) [inline shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-metrics/shardingsphere-agent-metrics-api/src/test/java/org/apache/shardingsphere/agent/metrics/api/advice/DataSourceAdviceTest.java:33]: It is suggested to be a final one.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-metrics/shardingsphere-agent-metrics-api/src/test/java/org/apache/shardingsphere/agent/metrics/api/advice/DataSourceAdviceTest.java (hunk 3 above)
Comment (tristaZero) [reply shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-metrics/shardingsphere-agent-metrics-api/src/test/java/org/apache/shardingsphere/agent/metrics/api/advice/DataSourceAdviceTest.java:33]: BTW, ci failed.

**Review by tristaZero — APPROVED** [review]
@dachuan9e Hi, good start, right?

**Comment by dachuan9e** [comment]
Releated to https://github.com/apache/shardingsphere/issues/11451

**Pull request merged**